rust_snuba/rust_arroyo/src/processing/strategies/reduce.rs

```diff
@@ -0,0 +1,201 @@
+use crate::processing::strategies::{CommitRequest, MessageRejected, ProcessingStrategy};
+use crate::types::{AnyMessage, InnerMessage, Message, Partition};
+use std::collections::BTreeMap;
+use std::time::{Duration, SystemTime};
+
+struct BatchState<T, TResult> {
+    value: Option<TResult>,
+    accumulator: fn(TResult, T) -> TResult,
```

Comment:
```suggestion
struct BatchState<T, TResult, F: FnMut(TResult, T) -> TResult> {
    value: Option<TResult>,
    accumulator: F,
```

Reply:
you can also make `accumulator: Box<dyn FnMut(...) -> ...>` if it's easier, then the generic is not needed on the struct, but only in `new`

this is probably better if people want to write `BatchState<T, TResult>`, and they can't if there's a closure being passed.